src/new_index/db_metrics.rs

```diff
@@ -0,0 +1,225 @@
+            ), labels),
+            num_entries_active_mem_table: metrics.gauge_vec(MetricOpts::new(
+                "rocksdb_num_entries_active_mem_table",
+                "Number of entries in active memtable"
+            ), labels),
+            num_entries_imm_mem_tables: metrics.gauge_vec(MetricOpts::new(
+                "rocksdb_num_entries_imm_mem_tables",
+                "Number of entries in immutable memtables"
+            ), labels),
+            num_deletes_active_mem_table: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_num_deletes_active_mem_table"),
+                "Number of delete entries in active memtable"
+            ), labels),
+            num_deletes_imm_mem_tables: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_num_deletes_imm_mem_tables"),
+                "Number of delete entries in immutable memtables"
+            ), labels),
+
+            // Compaction metrics
+            compaction_pending: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_compaction_pending"),
+                "Number of pending compactions"
+            ), labels),
+            estimate_pending_compaction_bytes: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_estimate_pending_compaction_bytes"),
+                "Estimated bytes pending compaction"
+            ), labels),
+            num_running_compactions: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_num_running_compactions"),
+                "Number of currently running compactions"
+            ), labels),
+            num_running_flushes: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_num_running_flushes"),
+                "Number of currently running flushes"
+            ), labels),
+
+            // Error metrics
+            background_errors: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_background_errors_total"),
+                "Total number of background errors"
+            ), labels),
+
+            // Key and data size estimates
+            estimate_num_keys: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_estimate_num_keys"),
+                "Estimated number of keys"
+            ), labels),
+            estimate_live_data_size: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_estimate_live_data_size_bytes"),
+                "Estimated live data size in bytes"
+            ), labels),
+            estimate_oldest_key_time: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_estimate_oldest_key_time_seconds"),
+                "Estimated oldest key timestamp"
+            ), labels),
+
+            // Table reader memory
+            estimate_table_readers_mem: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_estimate_table_readers_mem_bytes"),
+                "Estimated memory used by table readers in bytes"
```

Comment:
estimated memory used for reading SST tables, excluding memory used in block cache